### PR TITLE
Api integration

### PR DESCRIPTION
**Pull request by tomas-schmidt**
## Summary

- Add api integration with https://wbooks-api-stage.herokuapp.com to register users.

## Trello Card

- https://trello.com/c/Kej0fvlP/9-registration-api-integration

### CODE REVIEW

File: wolox-books/src/app/components/user/user.service.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  providedIn: 'root'
+})
+export class UserService {
+  readonly ROOT_URL = 'https://wbooks-api-stage.herokuapp.com/api/v1';
+  private header = { headers: new HttpHeaders().set('Content-type', 'application/json') };
+
+  constructor(private http: HttpClient) { }
+
+  createUser(user) {
+    console.log(user);
```
Comment (mcallegari10) [inline wolox-books/src/app/components/user/user.service.ts]: Delete console log 😄

File: wolox-books/src/app/components/user/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    this.http.post(this.ROOT_URL + "/users", user, this.header )
```
Comment (mcallegari10) [inline wolox-books/src/app/components/user/user.service.ts]: Use string interpolation for the endpoint

File: wolox-books/src/app/components/user/user.service.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+  readonly ROOT_URL = 'https://wbooks-api-stage.herokuapp.com/api/v1';
+  private header = { headers: new HttpHeaders().set('Content-type', 'application/json') };
+
+  constructor(private http: HttpClient) { }
+
+  createUser(user) {
```
Comment (mcallegari10) [inline wolox-books/src/app/components/user/user.service.ts]: This method should return either an `Observer` or a `Promise`

File: wolox-books/src/app/components/user/user.service.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    console.log(user);
+    this.http.post(this.ROOT_URL + "/users", user, this.header )
+      .subscribe(
```
Comment (mcallegari10) [inline wolox-books/src/app/components/user/user.service.ts]: Keep in mind that each endpoint can be used in different parts of the app, so you shouldn't subscribe to the response in the service method

File: wolox-books/src/app/components/user/user.ts [hunk 5]
```diff
@@ -1,8 +1,8 @@
 export interface User {
-    firstName: string;
-    lastName: string;
+    first_name: string;
+    last_name: string;
```
Comment (mcallegari10) [inline wolox-books/src/app/components/user/user.ts]: Why this changes? I think this is not really needed and you should use `camelCase` for naming

File: wolox-books/src/app/screens/register/register.component.ts [hunk 6]
```diff
@@ -27,6 +28,7 @@ export class RegisterComponent {
   }
 
   registerUser() {
-    console.log(JSON.stringify({ user: this.registerForm.value }));
+    const userJson = JSON.stringify({ user: this.registerForm.value });
```
Comment (mcallegari10) [inline wolox-books/src/app/screens/register/register.component.ts]: I think this is not needed, why is this stringified? 🤔

File: wolox-books/src/app/screens/register/register.component.ts [hunk 7]
```diff
@@ -27,6 +28,10 @@ export class RegisterComponent {
   }
 
   registerUser() {
-    console.log(JSON.stringify({ user: this.registerForm.value }));
+    this.userService.createUser({ user: this.registerForm.value })
+    .subscribe(
+      resp => { console.log('success') },
+      err => { }
```
Comment (mcallegari10) [inline wolox-books/src/app/screens/register/register.component.ts]: I think if error is blank, you should delete it

File: wolox-books/src/app/components/custom-validations/confirm.password.ts [hunk 8]
```diff
@@ -3,12 +3,12 @@ import {AbstractControl} from '@angular/forms';
 export class ConfirmPassword {
   static MatchPassword(AC: AbstractControl) {
     let password = AC.get('password').value;
-    let confirmPassword = AC.get('confirmPassword').value;
+    let confirmPassword = AC.get('confirm_password').value;
```
Comment (mcallegari10) [inline wolox-books/src/app/components/custom-validations/confirm.password.ts]: Why are the variables now snake_case and not camelCase? 🤔

Comment (tomas-schmidt) [reply wolox-books/src/app/components/custom-validations/confirm.password.ts]: Changed variables name so they match JSON request keys
